packages/publisher/github/src/util/github.ts

```diff
@@ -53,9 +53,9 @@ export default class GitHub {
   static sanitizeName(name: string): string {
     return path
       .basename(name)
+      .replace(/[^\w.@+-]+/g, '.')
       .replace(/\.+/g, '.')
       .replace(/^\./g, '')
-      .replace(/\.$/g, '')
-      .replace(/[^\w.-]+/g, '-');
+      .replace(/\.$/g, '');
```

Comment:
The `sanitize filename` algorithm can be found in the source code of github cli:
https://github.com/cli/cli/blob/b07f955c23fb54c400b169d39255569e240b324e/pkg/cmd/release/upload/upload.go#L131-L153